fabtests/multinode/src/core.c

```diff
@@ -135,6 +140,17 @@ static int multi_setup_fabric(int argc, char **argv)
 	if (ret)
 		return ret;
 
+	if (fi->domain_attr->mr_mode & FI_MR_LOCAL) {
+		ret = fi_mr_reg(domain, rx_barrier,
+				FT_MAX_CTRL_MSG * pm_job.num_ranks,
+				FT_MSG_MR_ACCESS, 0, BARRIER_TAG, 0,
+				&mr_barrier, NULL);
```

Comment:
We don't need a separate barrier buffer.  We only need to send a 0-byte message that uses the new tag.  The send will match with the receive.  Also, the memory registration key field is ignored unless the buffer is used for remote RMA operations.  But moving to a 0-byte barrier removes the need for this change anyway.

You probably don't even need directed receive as long as barrier messages use a separate tag, but I'm okay with that change.

Reply:
Comment still applies.  We can use a 0-byte send for barrier.  There's nothing that we to register, since there's no actual buffer that we need to exchange or post.